src/webgpu/shader/execution/expression/call/builtin/cos.spec.ts

```diff
@@ -40,17 +40,17 @@ TODO(#792): Decide what the ground-truth is for these tests. [1]
     u.combine('inputSource', allInputSources).combine('vectorize', [undefined, 2, 3, 4] as const)
   )
   .fn(async t => {
-    // [1]: Need to decide what the ground-truth is.
-    const makeCase = (x: number): Case => {
-      return makeUnaryF32Case(x, Math.cos);
+    const makeCase = (n: number): Case => {
+      return makeUnaryF32IntervalCase(n, cosInterval);
     };
 
-    // Spec defines accuracy on [-π, π]
-    const cases = linearRange(-Math.PI, Math.PI, 1000).map(x => makeCase(x));
+    const cases: Array<Case> = [
+      // Well defined accuracy range
+      ...linearRange(-Math.PI, Math.PI, 1000),
 
-    const cfg: Config = t.params;
-    cfg.cmpFloats = absMatch(2 ** -11);
-    run(t, builtin('cos'), [TypeF32], TypeF32, cfg, cases);
+      ...fullF32Range(),
```

Comment:
I thought there were no precision guarantees outside of  [-π, π]?
I see `cosInterval` uses `F32Interval.infinite()` for values outside of these ranges - so are we just checking here that things don't catch fire?

Reply:
Exactly